pylint/checkers/variables.py

```diff
@@ -2402,21 +2412,28 @@ def _check_is_unused(
 
             self.add_message(message_name, args=name, node=stmt)
 
-    def _is_name_ignored(self, stmt, name):
+    def _is_name_ignored(
+        self, stmt: nodes.NodeNG, name: str
+    ) -> re.Pattern[str] | re.Match[str] | None:
```

Comment:
This is relying on arcane knowledge around how python ``and`` operator work (the like of which should be limited to python brain teaser imo). But thanks for typing it. Do you think it's fixable easily ?

Reply:
For now, not really. I actually mistyped it initially because I misremembered how the operator worked.

I think for now it is probably fine? I don't like touching this code without a specific need. There is so much going on in this file..